packages/contract-wrappers/test/erc721_wrapper_test.ts

```diff
@@ -0,0 +1,460 @@
+        contractWrappers = new ContractWrappers(provider, config);
+        userAddresses = await web3Wrapper.getAvailableAddressesAsync();
+        tokens = tokenUtils.getDummyERC721TokenAddresses();
+        tokenAddress = tokens[0];
+        [ownerAddress, operatorAddress, anotherOwnerAddress, approvedAddress, receiverAddress] = userAddresses;
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    describe('#transferFromAsync', () => {
+        it('should fail to transfer NFT if fromAddress has no approvals set', async () => {
+            const tokenId = await tokenUtils.mintDummyERC721Async(tokenAddress, ownerAddress);
+            return expect(
+                contractWrappers.erc721Token.transferFromAsync(tokenAddress, receiverAddress, approvedAddress, tokenId),
+            ).to.be.rejectedWith(ContractWrappersError.ERC721NoApproval);
+        });
+        it('should successfully transfer tokens when sender is an approved address', async () => {
+            const tokenId = await tokenUtils.mintDummyERC721Async(tokenAddress, ownerAddress);
+            let txHash = await contractWrappers.erc721Token.setApprovalAsync(tokenAddress, approvedAddress, tokenId);
+            await web3Wrapper.awaitTransactionSuccessAsync(txHash);
+            const owner = await contractWrappers.erc721Token.getOwnerOfAsync(tokenAddress, tokenId);
+            expect(owner).to.be.equal(ownerAddress);
+            txHash = await contractWrappers.erc721Token.transferFromAsync(
+                tokenAddress,
+                receiverAddress,
+                approvedAddress,
+                tokenId,
+            );
+            await web3Wrapper.awaitTransactionSuccessAsync(txHash);
+            const newOwner = await contractWrappers.erc721Token.getOwnerOfAsync(tokenAddress, tokenId);
+            expect(newOwner).to.be.equal(receiverAddress);
+        });
+        it('should successfully transfer tokens when sender is an approved operator', async () => {
+            const tokenId = await tokenUtils.mintDummyERC721Async(tokenAddress, ownerAddress);
+            const isApprovedForAll = true;
+            let txHash = await contractWrappers.erc721Token.setApprovalForAllAsync(
+                tokenAddress,
+                ownerAddress,
+                operatorAddress,
+                isApprovedForAll,
+            );
+            await web3Wrapper.awaitTransactionSuccessAsync(txHash);
+            const owner = await contractWrappers.erc721Token.getOwnerOfAsync(tokenAddress, tokenId);
+            expect(owner).to.be.equal(ownerAddress);
+            txHash = await contractWrappers.erc721Token.transferFromAsync(
+                tokenAddress,
+                receiverAddress,
+                operatorAddress,
+                tokenId,
+            );
+            await web3Wrapper.awaitTransactionSuccessAsync(txHash);
```

Comment:
For all `awaitTransactionSuccessAsync` make sure to pass in `0` as the second parameter so tests run faster!